File: Codeforces/1114B.py
def make_string(ans):
    sss = ""
    for index, value in enumerate(ans):
        if(index):
            sss += " "
        sss += str(value)
    return sss


def each():
    n, m, k = [int(x) for x in input().split(" ")]
    arr = [[int(value), index]
           for index, value in enumerate(input().split(" "))]
    arr.sort()
    sum = 0
    len = (n-min(m*k, n))-1
    for i in range(n-1, len, -1):
        sum += arr[i][0]
        arr[i][0] = None
    arr.sort(key=lambda x: x[1:2])
    mmm = m
    ans = []
    for i in range(n):
        if(arr[i][0] == None):
            mmm -= 1
        if(mmm == 0):
            mmm = m
            ans.append(arr[i][1]+1)
    ans.pop()
    print(sum)
    ans = make_string(ans)
    print(ans)


each()

File: Codeforces/1114B.cpp
#include <bits/stdc++.h>
using namespace std;
#define not_valid 1000000001
typedef pair<int, int> ppi;
bool comp(ppi &A, ppi &B)
{
    return A.second < B.second;
}
int main()
{
    int n, m, k;
    scanf("%d%d%d", &n, &m, &k);
    vector<ppi> arr(n);
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &arr[i].first);
        arr[i].second = i;
    }
    sort(arr.begin(), arr.end(), greater<>());
    long long sum = 0;
    vector<int> ans;
    for (int i = 0; i < min(n, m * k); i++)
    {
        sum += arr[i].first;
        arr[i].first = not_valid;
    }
    sort(arr.begin(), arr.end(), comp);
    for (int i = 0, mmm = m; i < n; i++)
    {
        if (arr[i].first == not_valid)
            mmm--;
        if (mmm == 0)
        {
            mmm = m;
            ans.push_back(arr[i].second+1);
        }
    }
    ans.pop_back();
    printf("%lld\n", sum);
    for (auto it : ans)
    {
        printf("%d ", it);
    }
    puts("");
    return 0;
}